b9/core.cpp

```diff
@@ -352,24 +352,35 @@ void ExecutionContext::reset() {
   programCounter_ = 0;
 }
 
-StackElement VirtualMachine::run(const std::string& name) {
-  return run(module_->findFunction(name));
+StackElement VirtualMachine::run(const std::string& name, const std::vector<int>& usrArgs) {
+  return run(module_->findFunction(name), usrArgs);
 }
 
-StackElement VirtualMachine::run(const std::size_t functionIndex) {
+StackElement VirtualMachine::run(const std::size_t functionIndex, const std::vector<int>& usrArgs) {
   executionContext_.reset();
 
-  /* Push random arguments to send to the program */
   auto f = &module_->functions[functionIndex];
-
+  
+  // get/dispay function name and num arguments
   if (cfg_.verbose)
     std::cout << "function: " << functionIndex << " nargs: " << f->nargs
               << std::endl;
 
-  for (std::size_t i = 0; i < f->nargs; i++) {
-    int arg = 100 - (i * 10);
-    std::cout << "Pushing arg[" << i << "] = " << arg << std::endl;
-    executionContext_.push(arg);
+  if (usrArgs.size() == 0) {
+    // push random arguments to send to the program
```

Comment:
don't push random args, just delete this branch,

Reply:
@rwy0717 what if the user doesn't specify arguments for a function the requires them? error out?

Reply:
Yep, throw an exception, and catch it in b9run.